web/models/build_status_updater.ex

```diff
@@ -1,9 +1,27 @@
 defmodule Bigben.BuildStatusUpdater do
+  alias Bigben.Branch
   alias Bigben.Build
   alias Bigben.Repo
 
   def update(build_params) do
-    changeset = Build.changeset(%Build{}, build_params)
+    branch = get_branch(build_params)
```

Comment:
get_branch is a bit misleading because it gets or inserts. So maybe this should be `get_or_insert_branch`? What do you think?